src/Uno.UWP/ApplicationModel/PackageVersion.cs

```diff
@@ -1,6 +1,8 @@
+using System;
+
 namespace Windows.ApplicationModel
 {
-	public partial struct PackageVersion
+	public partial struct PackageVersion : IEquatable<PackageVersion>
```

Comment:
😄

Reply:
I added the interface here because that's what I found when in the decompiled sources from Windows head.

![image](https://user-images.githubusercontent.com/31348972/188613334-f27637c7-258a-4ccb-9c95-96a1ff636b86.png)

You'll find also that I just copied what I found in the decompiled sources regarding how `Equals` methods and `==` are implemented.

Also their `GetHashCode` is very prune to collisions, but I copied it as-is too.
It's probably fine to diverge from WinUI implementation here, what do you think?

Reply:
I wonder if `PackageVersion` should be a `record`, as it would create most of it automatically?

Reply:
> It's probably fine to diverge from WinUI implementation here, what do you think?

Yes. Code should not depend on this level of implementation details. It's part of why net6 added `HashCode.Combine` (and friends) to ensure you can't expect the same values between runs.

Reply:
I could switch to `record`, though I'll either have to do manual changes to generated files (which I don't like), or modify UWP sync generator to add `record` as well (though I don't know if it should add it for all structs?)

Reply:
Ok, let's not modify the generator for this particular case. Adding the suggestions for linker friendliness is still a good change.